File: backend/app/services/documents.py
from app.services.llm import extract_document_data, extract_image_data
from app.database import get_supabase
import base64


async def process_document_upload(document_id: str, file_content: str, doc_type: str, is_image: bool = False):
    try:
        if is_image:
            result = await extract_image_data(file_content, doc_type)
        else:
            result = await extract_document_data(file_content, doc_type)

        supabase = get_supabase()
        supabase.table("documents").update({
            "ai_extracted_data": result,
        }).eq("id", document_id).execute()

        return result
    except Exception as e:
        return {"error": str(e)}

File: backend/src/services/documents.ts
import { supabase } from "../database";
import { extractDocumentData, extractImageData } from "./llm";

export async function processDocumentUpload(
  documentId: string,
  fileContent: string,
  docType: string,
  isImage: boolean = false,
): Promise<Record<string, any>> {
  try {
    const result = isImage
      ? await extractImageData(fileContent, docType)
      : await extractDocumentData(fileContent, docType);

    await supabase
      .from("documents")
      .update({
        ai_extracted_data: result,
      })
      .eq("id", documentId);

    return result;
  } catch (e: any) {
    return { error: e.message };
  }
}
